src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -320,6 +323,61 @@ private static string GetExpectedRedirectUri(string clientId)
         }
 
         /// <summary>
+        /// Converts to MSAL Account Id or Null
+        /// </summary>
+        /// <param name="wamAccount"></param>
+        /// <param name="clientID"></param>
+        /// <param name="logger"></param>
+        public static IAccount ConvertToMsalAccount(
+                NativeInterop.Account wamAccount, 
+                string clientID,
+                ILoggerAdapter logger)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(wamAccount.AccountId) ||
+                    string.IsNullOrEmpty(wamAccount.HomeAccountid) ||
+                    string.IsNullOrEmpty(wamAccount.Environment) ||
+                    string.IsNullOrEmpty(wamAccount.UserName))
+                {
+                    // Create PII enabled string builder
+                    var builder = new StringBuilder(
+                        Environment.NewLine + "=== [WamBroker] Converting WAM Account to MSAL Account ===" + Environment.NewLine);
+
+                    builder.AppendLine($"wamAccount.AccountId: {wamAccount.AccountId}.");
+                    builder.AppendLine($"wamAccount.HomeAccountid: {wamAccount.HomeAccountid}.");
+                    builder.AppendLine($"wamAccount.UserName: {wamAccount.UserName}.");
+
+                    string messageWithPii = builder.ToString();
+
+                    // Create non PII enabled string builder
+                    builder = new StringBuilder(
+                        Environment.NewLine + "=== [WamBroker] Converting WAM Account to MSAL Account ===" + Environment.NewLine);
+
+                    builder.AppendLine($"wamAccount.AccountId: {string.IsNullOrEmpty(wamAccount.AccountId)}.");
+                    builder.AppendLine($"wamAccount.HomeAccountid: {string.IsNullOrEmpty(wamAccount.HomeAccountid)}.");
+                    builder.AppendLine($"wamAccount.Environment: {wamAccount.Environment}.");
+                    builder.AppendLine($"wamAccount.UserName: {string.IsNullOrEmpty(wamAccount.UserName)}.");
+
+                    logger.InfoPii(messageWithPii, builder.ToString());
+                    logger.Error($"[WamBroker] WAM Account properties are missing. Cannot convert to MSAL Accounts.");
+                    
+                    throw new MsalServiceException("wam_failed", $"WAM Account properties are missing.");
+                }
+
+                return new Account(
+                    wamAccount.HomeAccountid,
+                    wamAccount.UserName,
+                    wamAccount.Environment,
+                    new Dictionary<string, string>() { { clientID, wamAccount.AccountId } });
+            }
+            catch (Exception ex)
```

Comment:
I don't think we shold try / catch here. Just let the exception bubble up so that app developers can catch it and log bugs on MSAL.

Generally there are 2 patterns of dealing with exception: 

```csharp
// 1
IAccount ConvertToMsalAccount(WamAccount a) 
{
    // do the conversion; throw exceptions if WamAccount obj is in bad state
}

// 2 - see https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tryparse?view=net-7.0#system-datetime-tryparse(system-string-system-iformatprovider-system-globalization-datetimestyles-system-datetime@)

bool TryConvertToMsalAccount(WamAccount a, out IAccount acc)
{
     // do the conversion do not throw exceptions
    // if it fails, return false a
    // otherwise return true and set acc object

   // you can also take in a logger to log potential issues
}
```

It seems to me that option 2 is more approapriate here.